test_scripts/Polices/External_UCS/016_ATF_N_TC_PTU_with_external_consent_status_groups_struct.lua

```diff
@@ -0,0 +1,127 @@
+---------------------------------------------------------------------------------------------
+-- Requirement summary:
+-- [Policies] External UCS: PTU with "external_consent_status_groups" struct
+--
+-- Description:
+-- In case:
+-- SDL receives PolicyTableUpdate with "external_consent_status_groups:
+-- [<functional_grouping>: <Boolean>]” -> of "device_data" -> "<device identifier>"
+-- -> "user_consent_records" -> "<app id>" section
+-- SDL must:
+-- a. consider this PTU as invalid
+-- b. do not merge this invalid PTU to LocalPT.
+--
+-- Preconditions:
+-- 1. Modify PreloadedPT by adding 'disallowed_by_external_consent_entities_on' section
+--
+-- Steps:
+-- 1. Register app
+-- 2. Send SDL.OnAppPermissionConsent (change ON -> OFF)
+-- 3. Activate app
+-- 4. Perform PTU
+-- 5. Verify status of update
+--
+-- Expected result:
+-- PTU fails with UPDATE_NEEDED status
+--
+-- Note: Script is designed for EXTERNAL_PROPRIETARY flow
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+config.defaultProtocolVersion = 2
+
+--[[ Required Shared Libraries ]]
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local commonPreconditions = require('user_modules/shared_testcases/commonPreconditions')
+local json = require("modules/json")
+local testCasesForExternalUCS = require('user_modules/shared_testcases/testCasesForExternalUCS')
+
+--[[ Local variables ]]
+local appId = config.application1.registerAppInterfaceParams.appID
+local grpId = "Location-1"
+local checkedStatus = "UPDATE_NEEDED"
+
+--[[ Local Functions ]]
+local function replaceSDLPreloadedPtFile()
```

Comment:
@dboltovskyi, the idea of script is to verify that PTU will fail in case disallowed_by_external_consent_entities_on is observed in ptu file. I think that this function and actions related with it are redundant.